Movie test

jokirosdahl opened the pull request:
<!-- Thank you so much for your contribution! -->
<!-- First describe what your PR is doing.  -->
Test of the movie.f90 routine, added to the sink/stellar-HII test.
This is only for the (radiation-) hydro variables. Tests of particle movies are still to be done.

<!--
> [!WARNING]
>  Please uncomment this to tell everyone if there is a breacking change.
-->

<!-- Please check the following points -->
## PR Checklist

- [ ] The pull request has a title and a description
- [ ] For new code features, the PR should have been tested (please include the results of the tests below or, when possible, add a new test for the feature).
- [ ] The code covered by the pull request is documented

Commit:
Added testing of amr/movies.f90 to the sink/stellar-HII test.
Now the test plots three movie frames from the end of the test simulation and additionally checks the movie frames against a reference solution.

tinecolman commented on tests/visu/visu_ramses.py:
Thanks for adding the RT vars to visu_ramses.py!

tinecolman commented on tests/sink/stellar-HII/stellar-HII-ref.dat:
```diff
@@ -2,18 +2,34 @@ boxlen : 2.5000000000000000e+02
-velocity_z : 2.9307257548488060e+05
+velocity_x : 2.2258682653792299e+05
+velocity_y : 2.2258796740974986e+05
+velocity_z : 2.9307257509248570e+05
```
Can you put back the original reference value for the variables that were already present in the old reference?

tinecolman commented on tests/sink/stellar-HII/plot-stellar-HII.py:
```diff
@@ -57,33 +47,61 @@
+vars=["density", "temperature", "photon_flux_03", "scalar_00", "scalar_01", "scalar_02"]
+units=[unit_d/MH,  unit_v**2 * 2.37 * MH /KB, unit_v, 1., 1., 1.]
+labels=["log(density [H/cc])", "log(temperature [K])", "log(photon 3 flux [cm-2 s-1])",
+             "log(xHII)", "log(xHII)", "log(xHII)"]
```
The labels on the second row are all the same?

tinecolman commented on tests/sink/stellar-HII/stellar-HII.nml:
Maybe add a comment specifying the movie is at an angle? I was a bit confused when I first looked at the image.

tinecolman commented:
[test_results.pdf](https://github.com/user-attachments/files/20841374/test_results.pdf)

The test is failing when I run it on my laptop. Do the tolerance values need to be increased or could there be something wrong? It seems odd that flux3 is not 0.

jokirosdahl commented:
> [test_results.pdf](https://github.com/user-attachments/files/20841374/test_results.pdf)
> 
> The test is failing when I run it on my laptop. Do the tolerance values need to be increased or could there be something wrong? It seems odd that flux3 is not 0.

It's commented somewhere in the test script that it works only for 1 cpu, and it looks like you were running on two.

There are some random numbers being drawn in the create_stellar routine in this test, so the results will change with the number of CPUs, especially for the radiation variables, which have much higher values around the sinks (whereas other variables have only tiny variations around the sinks). I've therefore reduced the tolerance a lot for these variables.

Commit:
Fixes for recent commit adding movie.f90 to the stellar-HII test:
-increased tolerance for radiation quantities. There are small differences in the runs when run on varying numbers of mpi threads, due to random numbers being drawn for stellar sink particles. These lead to exagerrated differences in the radiation variables and ionisation fractions, because these are focused towards relatively very high values around the stellar sinks, whereas other variables are more or less homogeneous thoughout the whole box, and only have small variations from the homogeneous value around the sinks.
-updated reference values, now for 4 mpi threads
-fixed labels for plots of ionisation fractions
-comment in the namelist about the movie frames being at an angle

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

nbrucy commented:
The tests on Github are all run with 2 cpus.

tinecolman committed:
reset solution for old variables

tinecolman committed:
regenerate solution for 2 cpu

tinecolman committed:
adjust tolerances

tinecolman commented:
I've regenerated the solution on 2 cpus and adjusted the tolerances to make the test pass with 1 and 4 procs. And rebased @jokirosdahl dev branch...
